x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -117,7 +118,26 @@ public class TransportGetStackTracesAction extends HandledTransportAction<GetSta
     private final int desiredSlices;
     private final int desiredDetailSlices;
     private final boolean realtime;
-    private static final Map<String, HostMetadata> hostsTable = new ConcurrentHashMap<>();
+
+    /**
+     * The host metadata is needed to calculate the CO2 emission and the costs for each TraceEvent.
+     * We cache the host metadata to avoid querying the same host metadata multiple times and thus reduce API latency.
+     *
+     * TODO: Add timestamp to the HostMetadata and make the key a tuple of (hostID, timestamp).
+     *
+     * We just apply an upper limit to the number of cached hosts to ensure it doesn't blow up
+     * in unforeseen situations.
+     * The value itself is relatively arbitrary, but high enough to cover most foreseeable use cases.
+     */
+    private static final int MAX_HOSTS_CACHE_SIZE = 20_000;
```

Comment:
Since there is no time-based expiration, doesn't this cache risk not reflecting changes to the underlying host metadata that would affect the CO2 calculations? Can we update the doc string to mention how this is addressed (or not addressed).